src/pages/admin/profile/[user_id].tsx

```diff
@@ -0,0 +1,175 @@
+import { Suspense, useContext, useEffect, useState } from "react";
+
+import { NextRouter, useRouter } from "next/router";
+import { userContext } from "supabase/user_context";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+
+import { Avatar, Chip, Typography, Stack, Button, Box, TextField, FormHelperText } from "@mui/material";
+import MaleIcon from "@mui/icons-material/Male";
+import FemaleIcon from "@mui/icons-material/Female";
+import TransgenderIcon from "@mui/icons-material/Transgender";
+import CakeIcon from "@mui/icons-material/Cake";
+
+import Navbar from "@/components/admin/AdminNavbar";
+import Loading from "@/components/public/Loading";
+
+import { User } from "@/types/User";
+import { PagePaths } from "enum/pages";
+import { Gender } from "enum/gender";
+import { CHAR_LIMIT } from "enum/inputLimit";
+import { GetUserByUserId, UpdateUserNationalIdByUserId } from "@/services/User";
+import AdminVerifyDialog from "@/components/admin/AdminVerifyDialog";
+import { validateNationalIDCardNumber } from "@/utilities/validation";
+import { validation } from "@/types/Validation";
+import VerifyChip from "@/components/profile/VerifyChip";
+
+// style
+const profile_layout = {
+  width: "30vw",
+  minWidth: "200px",
+};
+const avatar = { width: 200, height: 200 };
+
+export default function adminProfile() {
+  const router: NextRouter = useRouter();
+  const userStatus = useContext(userContext);
+  const supabaseClient = useSupabaseClient<Database>();
+
+  const [targetUserData, setTargetUserData] = useState<User | null>(null);
+  const [nationalIDCard, setnationalIDCard] = useState<string>("");
+
+  const [isVerifyModalShow, setIsVerifyModalShow] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  useEffect(() => {
+    async function getTargetUserData() {
+      if (!userStatus.user || !router.query.user_id || targetUserData) return;
+      const userData = await GetUserByUserId(router.query.user_id as string, supabaseClient);
+      setTargetUserData(userData);
+    }
+
+    getTargetUserData();
+  }, [router.query.user_id, supabaseClient, userStatus.user, targetUserData]);
+
+  function verifyUser(): void {
+    console.log("confirm button is clicked!!", nationalIDCard);
+
+    const validate: validation = validateNationalIDCardNumber(nationalIDCard);
+    if (validate.err) {
+      setIsError(true);
+      setErrMsg(validate.msg);
+      return;
+    }
+
+    UpdateUserNationalIdByUserId(router.query.user_id as string, nationalIDCard, supabaseClient)
+      .then((is_national_id_exist) => {
+        if (is_national_id_exist) {
+          setIsError(true);
+          setErrMsg("เลขบัตรประจำตัวประชาชนนี้ถูกใช้งานไปแล้ว");
+          return;
+        }
+        router.reload()
+      })
+      .catch((err) => {
+        console.log(err);
+      });
```

Comment:
Bacause error message is the component of the dialog , and when error occurs, dialog still displays, should this one be included in AdminVerifyDialog or here?